pkg/templates/nstemplatetiers/nstemplatetier_generator.go

```diff
@@ -52,3 +40,47 @@ func CreateOrUpdateResources(ctx context.Context, s *runtime.Scheme, client runt
 		return applyCl.ApplyObject(ctx, toEnsure, commonclient.ForceUpdate(true))
 	}, namespace, metadata, files)
 }
+
+// LoadFiles takes the file from deploy/nstemplatetiers/<tiername>/<yaml file name> . the folder structure should be 4 steps .
+// as the cologic here is written accordingly
+func LoadFiles(nsTemplateTiers embed.FS, root string) (metadata map[string]string, files map[string][]byte, err error) {
+	// load templates from assets
+	metadataContent, err := nsTemplateTiers.ReadFile(root + "/metadata.yaml")
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "unable to load templates")
+	}
+
+	metadata = make(map[string]string)
+	err = yaml.Unmarshal(metadataContent, &metadata)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "unable to load templates")
+	}
+
+	paths, err := templates.GetAllFileNames(&nsTemplateTiers, root)
+	if err != nil {
+		return nil, nil, err
+	}
+	if len(paths) == 0 {
+		return nil, nil, fmt.Errorf("could not find any ns templates")
+	}
+	files = make(map[string][]byte)
+	for _, name := range paths {
+		if name == root+"/metadata.yaml" {
+			continue
+		}
+
+		parts := strings.Split(name, "/")
+		// skip any name that does not have 4 parts
+		if len(parts) != 4 {
+			return nil, nil, fmt.Errorf("unable to load templates: invalid name format for file '%s'", name)
+		}
+
+		fileName := parts[2] + "/" + parts[3]
```

Comment:
minor - I was wondering if we could have this as part of the `GetAllFileNames` ( and maybe it rename it to `GetAllTemplateNames` ), but I guess the `nsTemplateTiers.ReadFile(name)` would than fail. 

Or if there's a way using `nsTemplateTiers.ReadDir("templates/nstemplatetiers")` to just pass the content of the directory, so that we don't have to worry about the 4 parts . As all the other logic requires 2 parts only, see : https://github.com/codeready-toolchain/toolchain-common/blob/c8834962f739648258091d29f9dfe916f9bdadcb/pkg/template/nstemplatetiers/nstemplatetier_generator.go#L150-L152

So ideally also this logic should require 2 parts and not 4. 
WDYT?

Reply:
@mfrancisc as discussed i did give it a try.. the `.ReadDir()` returns the name object as 3 parts. so probably it will be almost the same logic as now. 
i do understand there might be some extra layer of logic as in toolchain-common with the current code. but i didnt want to change the logic in toolchain common, and hence did the change here.

Reply:
Thanks for looking into it. Yeah I agree it's fine to keep it as is for now, and later refactor the toolchain common logic maybe.